projects/lesson-7/runtime/src/kitties.rs

```diff
@@ -0,0 +1,464 @@
+use support::{
+	decl_module, decl_storage, decl_event, ensure, StorageValue, StorageMap,
+	Parameter, traits::Currency
+};
+use sr_primitives::traits::{SimpleArithmetic, Bounded, Member};
+use codec::{Encode, Decode};
+use runtime_io::blake2_128;
+use system::ensure_signed;
+use rstd::result;
+use crate::linked_item::{LinkedList, LinkedItem};
+
+
+// for impl encode and decode
+
+use support::dispatch::Output;
+use support::dispatch::Input;
+use codec::Error;
+
+// use arrayvec::ArrayVec;
+
+
+pub trait Trait: system::Trait {
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+	type KittyIndex: Parameter + Member + SimpleArithmetic + Bounded + Default + Copy;
+	type Currency: Currency<Self::AccountId>;
+}
+
+type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
+
+// #[derive(Encode, Decode)]
+#[derive(Debug, PartialEq)]
+pub struct Kitty(pub [u8; 16]);
+
+impl Encode for Kitty {
+	fn encode_to<W>(&self, dest: &mut W) where W:Output {
+		// for item in self.iter() {
+		// 	item.encode_to(dest);
+		// }
+		self.0.encode_to(dest);
+	}
+}
+
+impl Decode for Kitty {
+	fn decode<I>(input: &mut I) -> Result<Self, Error> where I: Input {
+		
+		match <[u8;16]>::decode(input) {
```

Comment:
Is it going to work? I think [u8;16] do have this implementation, but I haven't seen this way.